engine/runtime/src/main/java/org/enso/interpreter/runtime/type/Types.java

```diff
@@ -157,6 +154,10 @@ public static String getName(Object value) {
       return ConstantsGen.PANIC;
     } else if (TypesGen.isPanicSentinel(value)) {
       return ConstantsGen.PANIC;
+    } else if (TypesGen.isWarning(value)) {
+      return getName(TypesGen.asWarning(value).getValue());
+    } else if (value instanceof WithWarnings) {
+      return getName(((WithWarnings) value).getValue());
```

Comment:
Aren't these two branches redundant?

`TypesGen.isWarning` is implemented as `value instanceof WithWarnings`.

Reply:
Hm... I just checked and it is implemented as an instanceof [Warning](https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java) check
```java
    public static boolean isWarning(Object value) {
        return value instanceof Warning;
    }
```

Reply:
Ah, sorry I read that wrong indeed.

But in that case, shouldn't the first branch return `ConstantsGen.WARNING` instead? The `Warning` type is _not_ a transparent wrapper (like `WithWarnings`) - it contains the value as payload (see the `type Warning` in `Warning.enso`), so the type of the value present there is `Warning`, not the type of the contained payload.